crypto/src/lib.rs

```diff
@@ -26,11 +26,11 @@ pub use fastcrypto::traits;
 // to change all four aliases to point to concrete types that work with each other. Failure to do
 // so will result in a ton of compilation errors, and worse: it will not make sense!
 
-pub type PublicKey = ed25519::Ed25519PublicKey;
-pub type Signature = ed25519::Ed25519Signature;
-pub type AggregateSignature = ed25519::Ed25519AggregateSignature;
-pub type PrivateKey = ed25519::Ed25519PrivateKey;
-pub type KeyPair = ed25519::Ed25519KeyPair;
+pub type PublicKey = bls12381::BLS12381PublicKey;
```

Comment:
So, in theory this looks good, but in practice since #909 we also have worker keys. That worker key is used exclusively for the purpose of validating the authentication of workers over the network (TLS 1.3-powered), so it should morally be a `NetworkKey`

At the moment that key is defined in the field `name` of the `WorkerInfo` struct in `config`, which type is `PublicKey`. I think we need to switch `PublicKey` -> `NetworkingKey` there:
https://github.com/MystenLabs/narwhal/blob/a383ad626355035cf5e388be6ee668fff17bb646/config/src/lib.rs#L300-L302

Once that is done, note the changes needed to support worker keys in Sui are in https://github.com/MystenLabs/sui/pull/4541, so that the companion of the present PR (https://github.com/MystenLabs/sui/pull/4408) should be adapted to take 4541 into account.

Reply:
Done!